parser_test.go

```diff
@@ -1708,89 +1708,80 @@ var testJson = []byte(`{
 }`)
 
 func TestEachKey(t *testing.T) {
-	paths := [][]string{
-		{"name"},
-		{"order"},
-		{"nested", "a"},
-		{"nested", "b"},
-		{"nested2", "a"},
-		{"nested", "nested3", "b"},
-		{"arr", "[1]", "b"},
-		{"arrInt", "[3]"},
-		{"arrInt", "[5]"}, // Should not find last key
-		{"nested"},
-		{"arr", "["},    // issue#177 Invalid arguments
-		{"a\n", "b\n"},  // issue#165
-		{"nested", "b"}, // Should find repeated key
-		{"arrString", "[1]"},
+	tc := []struct {
+		path     []string
+		expected string
+		notFound bool
+	}{
+		{
+			path:     []string{"name"},
+			expected: "Name",
+		}, {
+			path:     []string{"order"},
+			expected: "Order",
+		}, {
+			path:     []string{"nested", "a"},
+			expected: "test",
+		}, {
+			path:     []string{"nested", "b"},
+			expected: "2",
+		}, {
+			path:     []string{"nested2", "a"},
+			expected: "test2",
+		}, {
+			path:     []string{"nested", "nested3", "b"},
+			expected: "4",
+		}, {
+			path:     []string{"arr", "[1]", "b"},
+			expected: "2",
+		}, {
+			path:     []string{"arrInt", "[3]"},
+			expected: "4",
+		}, {
+			path:     []string{"arrInt", "[5]"}, // Should not find last key
+			notFound: true,
+		}, {
+			path:     []string{"nested"},
+			expected: `{"a":"test", "b":2, "nested3":{"a":"test3","b":4}, "c": "unknown"}`,
+		}, {
+			path:     []string{"arr", "["}, // issue#177 Invalid arguments
+			notFound: true,
+		}, {
+			path:     []string{"a\n", "b\n"}, // issue#165
+			expected: "99",
+		}, {
+			path:     []string{"nested", "b"}, // Should find repeated key
+			expected: "2",
+		}, {
+			path:     []string{"arrString", "[1]"},
+			expected: "b",
+		}, {
+			path:     []string{"arrString", "[]"}, // [] is equal to [0]
```

Comment:
again, do we need `[]string{"arrString", "[\"\"]"}`, or maybe that case works?

Reply:
```
[]string{"keys", "[]"}
```
is what Loki produces when using this parser expression:

```
| json key[""]
```

---

```
[]string{"keys", "[\"\"]"}
```
would be equivalent to any other invalid, non-integer array index